contractcourt/commit_sweep_resolver_test.go

```diff
@@ -235,103 +241,52 @@ func TestCommitSweepResolverDelay(t *testing.T) {
 
 	ctx.waitForResult()
 
+	// If this test case generates a sweep error, we don't expect to be
+	// able to recover anything. This might happen if the local commitment
+	// output was swept by a justice transaction by the remote party.
+	expectedRecoveredBalance := btcutil.Amount(amt)
+	if sweepErr != nil {
+		expectedRecoveredBalance = 0
+	}
+
 	report = ctx.resolver.report()
-	if !reflect.DeepEqual(report, &ContractReport{
+	expectedReport = ContractReport{
 		Outpoint:         outpoint,
 		Type:             ReportOutputUnencumbered,
 		Amount:           btcutil.Amount(amt),
-		RecoveredBalance: btcutil.Amount(amt),
 		MaturityHeight:   testInitialBlockHeight + 2,
-	}) {
-		t.Fatal("unexpected resolver report")
-	}
-}
-
-// TestCommitSweepResolverLocalBreach tests resolution when the local node
-// publishes a breached output (one that is swept by the remote party).
-func TestCommitSweepResolverLocalBreach(t *testing.T) {
-	t.Parallel()
-	defer timeout(t)()
-
-	amt := int64(100)
-	outpoint := wire.OutPoint{
-		Index: 5,
-	}
-	res := lnwallet.CommitOutputResolution{
-		SelfOutputSignDesc: input.SignDescriptor{
-			Output: &wire.TxOut{
-				Value: amt,
-			},
-			WitnessScript: []byte{0},
-		},
-		MaturityDelay: 3,
-		SelfOutPoint:  outpoint,
-	}
-
-	ctx := newCommitSweepResolverTestContext(t, &res)
-
-	// Setup the sweeper so that it returns an error of the output being
-	// already swept.
-	ctx.sweeper.sweepErr = sweep.ErrRemoteSpend
-
-	report := ctx.resolver.report()
-	expectedReport := ContractReport{
-		Outpoint:     outpoint,
-		Type:         ReportOutputUnencumbered,
-		Amount:       btcutil.Amount(amt),
-		LimboBalance: btcutil.Amount(amt),
+		RecoveredBalance: expectedRecoveredBalance,
 	}
 	if *report != expectedReport {
```

Comment:
we should use `DeepEqual` here too.

Reply:
Just the deref is ok here since the struct only contains integers or fixed sized byte slices and has no pointer references.